assets/js/controllers/settings/info.controller.js

```diff
@@ -2,30 +2,29 @@ angular
   .module('walletApp')
   .controller('SettingsInfoCtrl', SettingsInfoCtrl);
 
-function SettingsInfoCtrl ($scope, Wallet, Alerts, $translate, $window) {
-  $scope.uid = Wallet.user.uid;
+function SettingsInfoCtrl ($scope, $q, Wallet, Alerts) {
+  angular.extend($scope, Wallet.user);
+  $scope.loading = {};
+  $scope.pairingCode = null;
 
-  $scope.display = {
-    pairingCode: false
+  $scope.removeAlias = () => {
+    $scope.loading.alias = true;
+    Alerts.confirm('CONFIRM_DISABLE_ALIAS', { id: $scope.guid })
+      .then(Wallet.removeAlias)
+      .then(() => $scope.alias = Wallet.user.alias, () => {})
+      .then(() => $scope.loading.alias = false);
```

Comment:
Yes, all errors are dealt with before that final `.then`, so `loading.alias` will always be set to `false` no matter what happens.